### DISCUSSION TITLE
On releasing unoriginal source code as a package

### DISCUSSION THREAD

**Discussion by me-aikawa** [sqlalchemy/sqlalchemy#9825] [Usage Questions] [answered]
Hi @CaselIT
This question relates to my previous question https://github.com/sqlalchemy/sqlalchemy/discussions/9378#discussion-4900854 here.

I took your advice and created a subclass named PGSpiderDialect. It only has a method that overrides the `_get_server_version_info` method of the `PGDialect_psycopg2` class. This allows me to connect to a DB built on PostgreSQL using SQLAlchemy.

```python
import re
from sqlalchemy.dialects.postgresql.psycopg2 import PGDialect_psycopg2

class PGSpiderDialect_psycopg2(PGDialect_psycopg2):
    supports_statement_cache = True

    def _get_server_version_info(self, connection):
        v = connection.exec_driver_sql("select pg_catalog.version()").scalar()
        m = re.match(
            r".*(?:PGSpider) "
            r"(\d+)\.?(\d+)?(?:\.(\d+))?(?:\.\d+)?(?:devel|beta)?",
            v,
        )
        if not m:
            raise AssertionError(
                "Could not determine version from string '%s'" % v
            )
        return tuple([int(x) for x in m.group(1, 2, 3) if x is not None])
```

I needed to package this code in order to use this subclass from certain software, so I created a new package named sqlalchemy-pgspider and used this in localspace.

Recently, I have had to share this package with several people. Those people have asked me to register this package with PyPI or Github. They expect to be able to install this package with `pip install`.

However, I feel that it is not a good idea to register this package with PyPI or GitHub. This is because most of the `_get_server_version_info methods` are copies of the same methods of the existing [sqlalchemy/dialects/postgresql.base.py](https://github.com/sqlalchemy/sqlalchemy/blob/main/lib/sqlalchemy/dialects/postgresql/base.py#L3195).

What do you think about this? How do you think I should act?
Sorry for the length of the question.
Your advice would be helpful.

**Comment by CaselIT** [comment, marked as answer]
hi

All the code of sqlalchemy is licensed under the MIT license, so it's not in general a problem to re-distribute parts or all it code in a 3rd party package.
IMNAL but I don't think there would be any issue in publishing that dialect on pypi and/or github.

**Reply by me-aikawa** [reply]
@CaselIT  Thank you for your comment.
I was missing a perspective on licensing.
I will now publish this package on GitHub and PyPI in the near future.
